doc/smart-switch/pmon/smartswitch-pmon.md

```diff
@@ -0,0 +1,281 @@
+* For the variable ones (temperature, voltage, fan speed ....) need to be collected periodically. 
+
+### 3.3. DPU STATE
+* DPU state represents the various operational states of the DPU. They are: Powered, PCIe-Link, Host-DPU Eth-Link, Firmware, OS, CP, DP.
+* The table below shows the various DPU states, the producer and consumers of them.
+#### 3.3.1. DPU_STATE - Before DPU OS boots
+* All DPU States are updated by the Host PMON
+* These states are available to host PMON as register bits from DPUs.
+* Additionally, the FPGA provides temp sensor values too
+#### 3.3.2. DPU_STATE - After DPU OS boots
+* DPU updates the DB with additional information such as the reason for every state, and timestamp 
+* Carries Die Temp and threshold
+* Has reason for previous reboot
+* Carries DP, CP up down reasons
+
+DPU_STATE Table
+
+| DPU_STATE | Powered | PCIe_Link | NPU_DPU_Ge_Link | Firmware | OS_boot |  | CtrlPlane | DataPlane |
+| --- | --- | --- | --- | --- | --- | --- | --- | --- |
+| DPU0 | on | up | up | up | up |  | up | down |
+| DPU1 | off | down | down | up | up |  | down | up |
+| Genrated By | dpu | dpu | dpu | dpu | dpu |  | dpu | dpu |
+| Updated By | ss-pmon | ss-pmon | ss-pmon | ss-pmon  | ss-pmon |  | ss-pmon | ss-pmon |
+
+Potential consumer: Switch CLIs, Utils (install/repair images), HA, LB, Life Cycle Manager 
+
+Use cases: Debuggability, error recovery (reset, power cycle) and fault management, consolidated view of Switch and DPU state/health
+
+#### 3.3.3. ChassisStateDB Schema for DPU_STATE
+```
+Table: “DPU_STATE”
+
+SCHEMA
+key:  dpu_state:1
+ 
+HMSET dpu_state:1
+        "id": "1",    		#Key itself can be used?
+        "powered": "ON",
+        "pcie_link_state": "UP",
+        "pcie_link_time": "timestamp",
+        "pcie_link_reason": "up_down_related string",
+        "host_eth_link_state": "UP",
+        "host_eth_link_time": " timestamp ",
+        "host_eth_link_reason": "up_down_related string",
+        "firmware_state": "UP",
+        "firmware_time": " timestamp ",
+        "firmware_reason": ”gold boot a, ONIE version x",
+        "os_state": "UP",
+        "os_state_time": "timestamp",
+        "os_reason": ”version x",
+        "previos_reboot_reason_from_dpu": “Software reboot ”,
+        "previos_reboot_time_from_dpu ": “timestamp”,
+        “previous_reboot_reason_from_host”: ”Powered Down DPU - Temperature failure”,
+        "previos_reboot_time_from_host ": “timestamp”,
+        "control_plane_state": ”DOWN",
+        "control_plane_time": ”timestamp",
+        "control_plane_reason": ”containers restarting",
+        "data_plane_state": ”DOWN",
+        "data_plane_time": ”timestamp",
+        "data_plane_reason": ”Pipeline failure",
+```
+
+Besides the state and previous_reboot_reason_from_host other fields will be updated by the DPU once it boots.  The other fields will be updated by the switch from the information read from the hardware registers.
+
+### 3.4.   Midplane Interface
```

Comment:
Hi @rameshraghupathy,

Smart Switch IP address assignment flow for midplane interface and DPU PCI interfaces is described [here](https://github.com/sonic-net/SONiC/blob/master/doc/smart-switch/ip-address-assigment/smart-switch-ip-address-assignment.md). Please check the document.

Reply:
Correct, let us align with above doc

Reply:
I have updated this with a new PR  https://github.com/sonic-net/SONiC/pull/1584.
Please look into this new PR.